src/components/Personalization/createExecuteDecisions.js

```diff
@@ -10,20 +10,30 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { assign } from "../../utils";
+import { assign, flatMap, isNonEmptyArray } from "../../utils";
+
+const identity = item => item;
 
 const buildActions = decision => {
-  const meta = { decisionId: decision.id };
+  const meta = { id: decision.id, scope: decision.scope };
 
   return decision.items.map(item => assign({}, item.data, { meta }));
 };
 
-export default ({ modules, logger, executeActions }) => {
+export default ({ modules, logger, executeActions, collect }) => {
   return decisions => {
-    decisions.forEach(decision => {
+    const decisionPromise = decisions.map(decision => {
       const actions = buildActions(decision);
 
-      executeActions(actions, modules, logger);
+      return executeActions(actions, modules, logger);
+    });
+    return Promise.all(decisionPromise).then(result => {
+      if (isNonEmptyArray(result)) {
+        const metas = flatMap(result, identity);
+        const decisionMetas = metas.map(item => item.meta);
+
+        collect({ decisions: decisionMetas });
```

Comment:
Hey @ninaceban, I think we should be returning the promise from `collect`. If the `collect` call fails, I think we want to be failing the `sendEvent` command. I think. You would also need to handle the promise coming out of the `executeDecisions` call in `createOnResponseHandler.js` so that we wait for the promise to finish allowing the `onResponse` lifecycle to complete.

Reply:
actually collect is a fire and forget action. I don't think onResponseHandler should wait for collect to complete. But we could add `then` and `catch` to collect call to at least log the success or failure that should be enough for debugging-troubleshooting.